app/models/concerns/url_validator.rb

```diff
@@ -0,0 +1,5 @@
+class UrlValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    record.errors.add attribute, "has invalid URL" if value !~ URI.regexp
```

Comment:
Make sure that url is only of images only like - gif, jpn, png

Reply:
Image URL ending with .gif, .jpg and .png is handled in Product Model with
```
validates :image_url, format: {
    with:   IMAGE_VALIDATION_REGEX,
    message: 'must be a URL for GIF, JPG, or PNG image.'
  }
```
The reason image file extension not handled in UrlValidator is that this validator can be used at other places to validate any URL